src/lib/chat/matrix-client.ts

```diff
@@ -70,6 +70,23 @@ export class MatrixClient implements IChatClient {
     return this.matrix.getAccountData(eventType);
   }
 
+  async getUserPresence(userId: string) {
+    await this.waitForConnection();
+
+    try {
+      const userPresenceData = await this.matrix.getPresence(userId);
+      const isOnline = userPresenceData?.currently_active || false;
```

Comment:
Oh yes my mistake. I didn't see that I could do `userPresenceData.presence`. I'll update that.

> Does the user record not have the same last_active_ago field or something similar.

in `mapUser` (non-async), `this.matrix.getUser` returns:
```
avatarUrl: undefined
currentlyActive: false
displayName: "95f5539c-4d0f-4f84-99d1-e0cda11fd006"
events: {}
lastActiveAgo: 0
lastPresenceTs: 0
modified: 1697204080849
presence: "offline"
presenceStatusMsg: undefined
rawDisplayName: "95f5539c-4d0f-4f84-99d1-e0cda11fd006"
userId: "@95f5539c-4d0f-4f84-99d1-e0cda11fd006:zero-synapse-develo
````

Like when we went through it earlier in the week, we saw that the lastActiveAgo and lastPresenceTs values will return the default `0` if not fetched asynchronously, and/or if the other user has been online since you've (current user [has]) been online. This was the reason for adding `getUserPresence` (async) method and updating the `lastSeenAt` in the saga.

Reply:
Does that mean the `presence` field we're using for whether they're online or not is also wrong?

Reply:
I guess in theory it is wrong, because the default for the others users `presence` will be `offline` if that user has not yet been online while you (current user) has been online. So I think it 'accidentally' works.

I could update and set `isOnline` presence data from `getUserPresence` in a follow up PR if it makes sense to do so, I'd say it does for accurate data right?

Reply:
Yeah, it seems like we should remove those fields from the mapUser and just rely on this. If it's unreliable in the one place there's no sense in pretending. That way they're both being set from the same spot.

Reply:
Makes sense! I'll do a follow up PR to update those. Thank youu